app/src/main/java/com/example/roomies/HomeFragment.java

```diff
@@ -44,6 +72,149 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         // Inflate the layout for this fragment
         View view = inflater.inflate(R.layout.fragment_home, container, false);
+
+        // initialize list of all UserCircles related to current circle
+        userCircleList = new ArrayList<>();
+
+        // bind layout
+        ivCirclePhoto = view.findViewById(R.id.ivCirclePhoto);
+        tvCircleName = view.findViewById(R.id.tvCircleName);
+        ivProfile1 = view.findViewById(R.id.ivProfile1);
+        ivProfile2 = view.findViewById(R.id.ivProfile2);
+        ivProfile3 = view.findViewById(R.id.ivProfile3);
+        ivProfile4 = view.findViewById(R.id.ivProfile4);
+        ivProfile5 = view.findViewById(R.id.ivProfile5);
+        tvExtraProfiles = view.findViewById(R.id.tvExtraProfiles);
+
+        getCircle(getActivity(), view);
+
         return view;
     }
+
+    /**
+     * Query UserCircle objects that contain current user to get circles that user has joined
+     */
+    public void getCircle(Context context, View view){
+
+        // specify what type of data we want to query - UserCircle.class
+        ParseQuery<UserCircle> query = ParseQuery.getQuery(UserCircle.class).whereEqualTo(UserCircle.KEY_USER, ParseUser.getCurrentUser());
+        // include data referred by user key
+        query.include(UserCircle.KEY_USER);
+        query.include(UserCircle.KEY_CIRCLE);
+        // start an asynchronous call for UserCircle objects that include current user
+        query.findInBackground(new FindCallback<UserCircle>() {
+            @Override
+            public void done(List<UserCircle> userCircles, ParseException e) {
+                // check for errors
+                if (e != null) {
+                    Log.e(TAG, "Issue with getting userCircles", e);
+                    return;
+                }
+
+                // user has not joined a circle
+                if(userCircles.isEmpty()){
+                    Intent i = new Intent(getActivity(), AddCircleActivity.class);
+                    startActivity(i);
+                    getActivity().finish();
+                }
+
+                // save received posts to list and notify adapter of new data
+                circle = userCircles.get(0).getCircle();
+
+                // fill image and text on home screen
+                Glide.with(context).load(circle.getImage().getUrl()).apply(RequestOptions.circleCropTransform()).into(ivCirclePhoto);
+                tvCircleName.setText(circle.getName());
+
+                getAllProfiles(view);
+            }
+        });
+    }
+
+    /**
+     * Get all UserCircles related to current circle
+     */
+    public void getAllProfiles(View view){
```

Comment:
Does this function need to be a public function? could it be a private function?